Tests fail with `MSI not available` when some env vars are missing

/kind bug

**What steps did you take and what happened:**

Run `make go-test`. I got some test failures, all of them stating the underlying error
```
MSI not available
```


**What did you expect to happen:**
Tests pass.

**Anything else you would like to add:**

Exporting these env vars seems to fix the problem
```
export AZURE_CLIENT_ID="client-id"
export AZURE_CLIENT_SECRET="something"
```


**Environment:**

- cluster-api-provider-azure version: master
- Kubernetes version: (use `kubectl version`): v1.19.7
- OS (e.g. from `/etc/os-release`):  Ubuntu 20.04.2 LTS

The errors are:

```
--- FAIL: TestAzureMachinePoolMachineReconciler_Reconcile (1.02s)
    --- FAIL: TestAzureMachinePoolMachineReconciler_Reconcile/should_successfully_reconcile (0.51s)
        azuremachinepoolmachine_controller_test.go:57:
            Unexpected error:
                <*errors.withStack | 0xc000788018>: {
                    error: <*errors.withMessage | 0xc000786000>{
                        cause: <*errors.errorString | 0xc0000b6010>{
                            s: "MSI not available",
                        },
                        msg: "failed to configure azure settings and credentials from environment",
                    },
                    stack: [0x238baa9, 0x2593238, 0x25b8657, 0x111f06f, 0x1073de1],
                }
                failed to configure azure settings and credentials from environment: MSI not available
            occurred
        panic.go:613: missing call(s) to *mocks.MockReconciler.Reconcile(expected a context.Context, but got <nil>) /Users/cerobert/go/src/sigs.k8s.io/cluster-api-provider-azure/exp/controllers/azuremachinepoolmachine_controller_test.go:53
        panic.go:613: aborting test due to missing call(s)
    --- FAIL: TestAzureMachinePoolMachineReconciler_Reconcile/should_successfully_delete (0.51s)
        azuremachinepoolmachine_controller_test.go:71:
            Unexpected error:
                <*errors.withStack | 0xc0007601e0>: {
                    error: <*errors.withMessage | 0xc000774c20>{
                        cause: <*errors.errorString | 0xc0004625e0>{
                            s: "MSI not available",
                        },
                        msg: "failed to configure azure settings and credentials from environment",
                    },
                    stack: [0x238baa9, 0x2593238, 0x25b8657, 0x111f06f, 0x1073de1],
                }
                failed to configure azure settings and credentials from environment: MSI not available
            occurred
        panic.go:613: missing call(s) to *mocks.MockReconciler.Delete(expected a context.Context, but got <nil>) /Users/cerobert/go/src/sigs.k8s.io/cluster-api-provider-azure/exp/controllers/azuremachinepoolmachine_controller_test.go:67
        panic.go:613: aborting test due to missing call(s)
 ```

--- a/exp/controllers/azuremachinepoolmachine_controller_test.go
+++ b/exp/controllers/azuremachinepoolmachine_controller_test.go
@@ -18,8 +18,11 @@ package controllers
 
 import (
 	"context"
+	"os"
 	"testing"
 	"time"
+
+	"github.com/Azure/go-autorest/autorest/azure/auth"
 
 	"github.com/golang/mock/gomock"
 	. "github.com/onsi/gomega"
@@ -72,6 +75,10 @@ func TestAzureMachinePoolMachineReconciler_Reconcile(t *testing.T) {
 			},
 		},
 	}
+
+	os.Setenv(auth.ClientID, "fooClient")
+	os.Setenv(auth.ClientSecret, "fooSecret")
+	os.Setenv(auth.TenantID, "fooTenant")
 
 	for _, c := range cases {
 		t.Run(c.Name, func(t *testing.T) {

